Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -523,13 +523,28 @@ For additional best practices and detailed information please refer to the follo
 
 ### Testing Login Activity and Device Blocking (MSTG-AUTH-11)
 
-For applications which require L2 protection, the MASVS states that: "The app informs the user of all login activities with their account. Users are able view a list of devices used to access the account, and to block specific devices.". This can be broken down into various scenarios:
+For applications which require L2 protection, the MASVS states that they should inform the user about all login activities within the app with the possiblity of blocking certain devices. This can be broken down into various scenarios:
 
-1. The application provides a push notification the moment their account is used on another device to notify the user of different activities. The user can then block this device after opening the app via the push-notification.
+1. The application provides a push notification the moment their account is used on another device to notify the user of different activities. The user can then block this device after opening the app via the push-notification. 
 2. The application provides an overview of the last session after login, if the previous session was with a different configuration (e.g. location, device, app-version) then the user his current configuration. The user then has the option to report suspicious activities and block devices used in the previous session.
 3. The application provides an overview of the last session after login at all times.
 4. The application has a self-service portal in which the user can see an audit-log and manage the different devices with which he can login.
 
+The developer can make use of specific meta-information and associate it to each different activity or event within the application. This will make it easier for the user to spot suspicious behavior and block the corresponding device. The meta-information may include:
+
+- Device: The user can clearly identify all devices where the app is being used.
+- Date and Time: The user can clearly see the latest date and time when the app was used.
+- Location: The user can clearly identify the lastest locations where the app was used.
+
+The application can provide a push notification after each sensitive activity within the application. The choice of which activities to audit needs to be done for each application based on the data it handles and the level of security risk the team is willing to have. Below is a list of common sensitive activies that are usually audited:
```

Comment:
This could also just be in the list of activities/account history as an alternative to push notifications :). (e.g. see more than just the last location, but more or less a list of locations used: it should be a balance between privacy and security depending on the risks involved)